### ISSUE TITLE
[sort-keys] Add option to allow shorthand in object literals to be first/last

### ISSUE THREAD

**Issue by zimme** [eslint/eslint#7543]
**Please show your configuration:**
```
'sort-keys': 'error',
```
**What did you do? Please include the actual source code causing the issue.**

```js
export const ActiveRoute = {
  config,
  registerRouter,
  name: isActiveRouteName,
  path: isActiveRoutePath,
};
```

**What did you expect to happen?**

I'd expect this not to error out as I prefer to have shorthand assignments before regular assignment, but I'd like them sorted.

**What actually happened? Please include the actual, raw output from ESLint.**

```
Expected object keys to in natural insensitive ascending order. 'name' shoud be before 'registerRouter'.`
```

**Comment by kaicataldo** [comment]
This seems like a reasonable request to me, but I have some questions. This would have to also account for shorthand methods and be an additional configuration option that is off by default. Perhaps `separateShorthand`?

**Comment by zimme** [comment]
I'm ok with this being an option and I don't see a problem with this option affecting all shorthand things, but in that case it would be nice if one could opt-in for even more granular control over which shorthands should sort in this way.

**Comment by zimme** [comment]
```
'sort-keys': ['error', { shorthand: 'first' }] // Would sort all shorthands before other keys
'sort-keys': ['error', { shorthand: 'last'}] // Would sort all shorthands after other keys
'sort-keys', ['error', { shorthand: { properties: 'first' } }] // Would sort all variable shorthands before other keys
'sort-keys': ['error', { shorthand: { functions: 'last' } }] // Would sort all function shorthands after other keys
```
Something like this maybe?

**Comment by kaicataldo** [comment]
I'm not convinced of the schema you suggested. What happens if it's configured as follows?

```
'sort-keys': ['error', { shorthand: { properties: 'first', functions: 'last' } }]
```
And if we're allowing granular control over that, I don't think it would make sense *not* to be able to do the same for regular properties/methods.

I was envisioning something more along the lines of a single `{ separateShorthand: true }` option that would allow/require shorthand and regular properties to be grouped and sorted by the other options specified, but not necessarily enforce which should come first. Something similar to the enhancement suggested [here](https://github.com/eslint/eslint/issues/6175).

**Comment by zimme** [comment]
I think just grouping and sorting shorthands separately is a good enough start.

**Comment by nzakas** [comment]
Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
